### PR TITLE
Fix/sgw mac support

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add mac support to run tests on mac OS

Jenkins job link which ran against Centos to show it did not break existing tests:
http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-base-cc-converg/1087/testReport/

Jenkins job linke which ran on MacOS

http://uberjenkins.sc.couchbase.com:8080/view/04%20sync-gateway/job/Macos-sync-gateway-functional-tests-base/18/testReport/
Failed tests passed with reran except maxBack ups tests which needs test fix accross all platforms

### CODE REVIEW

File: libraries/provision/ansible/playbooks/tasks/create-sync-gateway-user.yml [hunk 1]
```diff
@@ -1,7 +1,46 @@
...
+
+# - name: SYNC GATEWAY | Create group policy for sync gateway on MacOS
+#   become: yes
+#   shell: dseditgroup -o create sync_gateway
+#   when: ansible_distribution == "MacOSX"
+
+# - name: SYNC GATEWAY | Add Sync gateway user to sync gateway group on MacOS
+#   become: yes
+#   shell: dseditgroup -o edit -a sync_gateway -t user sync_gateway
+#   when: ansible_distribution == "MacOSX"
```
Comment (euniceatcouchbase) [inline libraries/provision/ansible/playbooks/tasks/create-sync-gateway-user.yml]: may these code still have a chance to be rollback for debugging convenience? if not, it's better to get rid of them.

File: libraries/provision/ansible/playbooks/tasks/install-sg.yml [hunk 2]
```diff
@@ -9,4 +9,10 @@
   shell: dpkg -i /tmp/{{ couchbase_sync_gateway_package }}
   when: ansible_distribution == "Ubuntu" or ansible_distribution == "Debian"
 
+- name: SYNC GATEWAY | Install sync_gateway tar.gz
+  become: yes
+  shell: |
+    tar -zxvf /tmp/{{ couchbase_sync_gateway_package }} --directory /opt
```
Comment (euniceatcouchbase) [inline libraries/provision/ansible/playbooks/tasks/install-sg.yml]: sorry, not a comment, but trying to learn and understand ...... what is the command before the pipe?

File: testsuites/syncgateway/functional/tests/test_log_redaction.py [hunk 3]
```diff
@@ -85,7 +85,7 @@ def test_log_redaction_config(params_from_base_test_setup, remove_tmp_sg_redacti
     ("log_redaction", "none", False),
     ("log_redaction", "partial", True)
 ])
-def test_sgCollect1(params_from_base_test_setup, remove_tmp_sg_redaction_logs, sg_conf_name, redaction_level, redaction_salt):
+def te_st_sgCollect1(params_from_base_test_setup, remove_tmp_sg_redaction_logs, sg_conf_name, redaction_level, redaction_salt):
```
Comment (euniceatcouchbase) [inline testsuites/syncgateway/functional/tests/test_log_redaction.py]: te_st? is this an accident?

File: libraries/provision/ansible/playbooks/tasks/check-sync-gateway-macos.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+- name: SYNC GATEWAY | check sync_gateway service status on Windows
+  win_command: sc query SyncGateway
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply libraries/provision/ansible/playbooks/tasks/check-sync-gateway-macos.yml]: good catch. Looks like I created all yml files which was written for other platforms. As this was not used anywhere, I did not make change for macos. I deleted the file now as we are not using this ansible anywhere for other platforms too.

File: libraries/provision/ansible/playbooks/tasks/create-sync-gateway-user.yml (hunk 1 above)
Comment (sridevi-15) [reply libraries/provision/ansible/playbooks/tasks/create-sync-gateway-user.yml]: As we discussed , I am leaving commented and added comment with reason why this is left as comment.

File: libraries/provision/ansible/playbooks/tasks/install-sg.yml (hunk 2 above)
Comment (sridevi-15) [reply libraries/provision/ansible/playbooks/tasks/install-sg.yml]: A command with pipe can be used if you have multiple shell commands. It combines all shell commands and run one after the other.

File: testsuites/syncgateway/functional/tests/test_log_redaction.py (hunk 3 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_log_redaction.py]: oops, I fixed it locally , but forgot to checkin. good catch.

**Pull request merged**